spring-integration-zmq/src/main/java/org/springframework/integration/zmq/inbound/ZmqMessageDrivenChannelAdapter.java

```diff
@@ -0,0 +1,326 @@
+/*
+ * Copyright 2002-2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.integration.zmq.inbound;
+
+import java.nio.charset.Charset;
+import java.util.Date;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledFuture;
+
+import org.zeromq.ZMQ;
+import org.zeromq.ZMQException;
+
+import org.springframework.context.ApplicationEventPublisher;
+import org.springframework.context.ApplicationEventPublisherAware;
+import org.springframework.integration.zmq.core.ConsumerStopAction;
+import org.springframework.integration.zmq.core.DefaultZmqClientFactory;
+import org.springframework.integration.zmq.core.ZmqClientFactory;
+import org.springframework.integration.zmq.event.ZmqConnectionFailedEvent;
+import org.springframework.integration.zmq.event.ZmqSubscribedEvent;
+import org.springframework.messaging.Message;
+import org.springframework.messaging.MessagingException;
+import org.springframework.util.Assert;
+
+/**
+ * Jeromq Implementation.
+ *
+ * @author Subhobrata Dey
+ *
+ * @since 5.1
+ *
+ */
+public class ZmqMessageDrivenChannelAdapter extends AbstractZmqMessageDrivenChannelAdapter
```

Comment:
Is there really a reason to extract an abstract class?
Maybe we just can live with the `jeromqp` support and that's all?